esp_jpeg/jpeg_decoder.c

```diff
@@ -119,6 +119,50 @@ esp_err_t esp_jpeg_decode(esp_jpeg_image_cfg_t *cfg, esp_jpeg_image_output_t *im
     return ret;
 }
 
+#define LDB_WORD(ptr)       (unsigned short)(((unsigned short)*((unsigned char*)(ptr))<<8)|(unsigned short)*(unsigned char*)((ptr)+1))
```

Comment:
Perhaps you can use a slightly more readable function
```
static inline uint16_t ldb_word(const void* ptr) {
  uint16_t result;
  memcpy(&result, ptr, sizeof(result);
  return result;
}
```

the compiler will optimize out memcpy replacing it with either an unaligned load or a pair of byte loads, depending on whether the CPU supports unaligned loads or not.

Reply:
@igrr I ended up with the following implementation to preserve the big-endian to little-endian transformation. All other comments have been incorporated!
``` c
static inline uint16_t ldb_word(const void *ptr)
{
    const uint8_t *p = (const uint8_t *)ptr;
    return ((uint16_t)p[0] << 8) | p[1];
}
```